codemon/codemon.py

```diff
@@ -18,26 +20,47 @@ def main():
       countArg+=1;
 
       if arg == "init":
+        if sys.argv[countArg] == '-cpp':
+          if sys.argv[countArg] == '-n':
+            file = sys.argv[countArg+1]
+            path = '.'
+            f = open(path + '/' + file + '.cpp',"w+")
+            template = template_cpp()
+            f.write(template)
+            f.close()
+            print(colored.yellow("Created "+file+'.cpp'))
+            break;
 
-        if sys.argv[countArg] == '-n':
-          file = sys.argv[countArg+1]
-          path = '.'
-          f = open(path + '/' + file + '.cpp',"w+")
-          template = template_cpp()
-          f.write(template)
-          f.close()
-          print(colored.yellow("Created "+file+'.cpp'))
-          break;
-
-        else:
-          contestName = sys.argv[countArg]
-          fileNames = get_filename()
-          init(contestName, fileNames)
+          else:
+            contestName = sys.argv[countArg]
+            fileNames = get_filename()
+            init(contestName, fileNames)
+
+        elif sys.argv[countArg] == '-java':
+          
+          if sys.argv[countArg] == '-n':
+            file = sys.argv[countArg+1]
+            path = '.'
+            f = open(path + '/' + file + '.java',"w+")
+            template = template_java()
+            f.write(template)
+            f.close()
+            print(colored.yellow("Created "+file+'.java'))
+            break;
+          else:
+            contestName = sys.argv[countArg]
+            fileNames = get_filename_java()
+            initjava(contestName, fileNames)
 
       elif arg == "listen":
         listen()
 
       elif arg == "practice":
-        contestName = sys.argv[countArg]
-        practiceFiles = get_practice_files()
-        init(contestName, practiceFiles)
+        if sys.argv[countArg] == '-cpp':
+          contestName = sys.argv[countArg]
+          practiceFiles = get_practice_files()
+          init(contestName, practiceFiles)
+        elif sys.argv[countArg] == '-java':
+          contestName = sys.argv[countArg]
+          practiceFiles = get_practice_files_java()
```

Comment:
Argument parsing is not proper ! If we type `codemon init -n newFile -java`, I get a directory created as "-java/". 
Also, `codemon init newDirectory -java` doesn't work. Same for C++ files. You need to refactor this function.